### ISSUE TITLE
No way to setUserId or setCustomData for reward videos

### ISSUE THREAD

**Issue by dodocodes** [admob-plus/admob-plus#196]
There is currently no way that I can find to [setCustomData](https://developers.google.com/android/reference/com/google/android/gms/ads/rewarded/ServerSideVerificationOptions.Builder#setCustomData(java.lang.String)) or [setUserId](https://developers.google.com/android/reference/com/google/android/gms/ads/rewarded/ServerSideVerificationOptions.Builder#setUserId(java.lang.String)) to verify reward videos server side using the reward video ssv: https://developers.google.com/admob/android/rewarded-video-ssv

This is essential for reward videos to verify that the user actually completed the video reward, rather than just relying on a client side callback to reward the user.

The [IAdRequest](https://admob-plus.github.io/docs/ad-request-options#iadrequest) interface currently supports the following data:
```typescript
interface IAdRequest {
  id?: AdUnitIDOption
  testDevices?: string[]
  childDirected?: boolean
  underAgeOfConsent?: boolean
}
```

Maybe allowing a customData object or a userId here that gets set using the setUserId or setCustomData functions would be the way to go?

For example:
```javascript
document.addEventListener('deviceready', () => {
  admob.rewardVideo.load({
    id: {
      // replace with your ad unit IDs
      android: 'ca-app-pub-xxx/yyy',
      ios: 'ca-app-pub-xxx/zzz',
    },
    userId: '14239023',
    customData: {
        somethingElse: 'hello world'
    }
  }).then(() => admob.rewardVideo.show())
}, false)
```

**Comment by bardgr** [comment]
Hugely useful in order to support reward vids and increase revenue. Many thumbs up.
Is it possible @ratson to have it in the next release?

**Comment by nfacha** [comment]
Also interested in this @ratson

**Comment by SeVeNDuS** [comment]
This is pretty useful @ratson 
Are this planned for v1?
Thanks

**Comment by nfacha** [comment]
This also applies for rewarded interstitials

**Comment by ratson** [comment]
@dodocodes @bardgr @nfacha @SeVeNDuS 
I have just released `admob-plus-cordova@1.0.0-alpha.13` to support this.
Please refer to https://admob-plus.github.io/docs/cordova/rewarded-ads-ssv/

**Comment by nfacha** [comment]
@ratson After updating production ads stopped showing (test ads show)
I get this on logcat, unsure if related:

```java
03-06 18:10:27.281 28702 28702 E AndroidRuntime: java.lang.ClassCastException: admob.plugin.ads.BannerAd cannot be cast to admob.plugin.ads.InterstitialAd
03-06 18:10:27.281 28702 28702 E AndroidRuntime: 	at admob.plugin.ads.InterstitialAd.getOrCreate(InterstitialAd.java:18)
03-07 12:28:07.905 23046 23046 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:28:07.905 23046 23046 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:28:07.905 23046 23046 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:28:07.905 23046 23046 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:28:07.905 23046 23046 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:30:14.412 25326 25326 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:30:14.412 25326 25326 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:30:14.412 25326 25326 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:30:14.412 25326 25326 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:30:14.412 25326 25326 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:30:19.570 25511 25511 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:30:19.570 25511 25511 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:30:19.570 25511 25511 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:30:19.570 25511 25511 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:30:19.570 25511 25511 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:30:26.676 25663 25663 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:30:26.676 25663 25663 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:30:26.676 25663 25663 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:30:26.676 25663 25663 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:30:26.676 25663 25663 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:31:02.920 26126 26126 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:31:02.920 26126 26126 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:31:02.920 26126 26126 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:31:02.920 26126 26126 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:31:02.920 26126 26126 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:31:30.572 26494 26494 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:31:30.572 26494 26494 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:31:30.572 26494 26494 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:31:30.572 26494 26494 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:31:30.572 26494 26494 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:45:40.086  5857  5857 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:45:40.086  5857  5857 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:45:40.086  5857  5857 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:45:40.086  5857  5857 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:45:40.086  5857  5857 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:45:45.602  6082  6082 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:45:45.602  6082  6082 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:45:45.602  6082  6082 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:45:45.602  6082  6082 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:45:45.602  6082  6082 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:47:45.607  7484  7484 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:47:45.607  7484  7484 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:47:45.607  7484  7484 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:47:45.607  7484  7484 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:47:45.607  7484  7484 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 12:47:50.700  7612  7612 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException: 
03-07 12:47:50.700  7612  7612 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:47:50.700  7612  7612 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
03-07 12:47:50.700  7612  7612 E AndroidRuntime: 	at com.google.android.gms.internal.ads.zzzt.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:24)
03-07 12:47:50.700  7612  7612 E AndroidRuntime: 	at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@19.7.0:3)
03-07 18:21:50.477 20443 20443 E AndroidRuntime: java.lang.ClassCastException: admob.plugin.ads.RewardedAd cannot be cast to admob.plugin.ads.InterstitialAd
03-07 18:21:50.477 20443 20443 E AndroidRuntime: 	at admob.plugin.ads.InterstitialAd.getOrCreate(InterstitialAd.java:18)
03-07 18:53:06.298 17292 17292 E AndroidRuntime: java.lang.ClassCastException: admob.plugin.ads.RewardedAd cannot be cast to admob.plugin.ads.InterstitialAd
03-07 18:53:06.298 17292 17292 E AndroidRuntime: 	at admob.plugin.ads.InterstitialAd.getOrCreate(InterstitialAd.java:18)

```
